sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/pythonLogicalOperators.scala

```diff
@@ -38,3 +38,30 @@ case class FlatMapGroupsInPandas(
    */
   override val producedAttributes = AttributeSet(output)
 }
+
+trait BaseEvalPython extends UnaryNode {
```

Comment:
Is `producedAttributes` missing from this? Previously, `BatchEvalPython` and `ArrowEvalPython` have it defined.

Reply:
This is a problem I want to address later. I think `producedAttributes` makes no sense. It's only used to define `missingInput`, but we can overwrite `reference` to do the same thing.

More specifically, if `reference` is wrong implemented, column pruning will be broken. If `producedAttributes` is not implemented, nothing seriously will happen.